### ISSUE TITLE
Form tag doesn't include the form.id

### ISSUE THREAD

**Issue by naxorn** [Shopify/liquid#1650]
# Issue
When creating a form such as
```liquid
{% form 'customer', id: 'ContactFooter' %}
  <span>{{ form.id }}</span>
  [...rest of content...]
{% endform %}
```
the ```form.id``` is ```nil```

# Expected
The form object ```id``` should be the id of the submitted form and be available within the form.  This is especially useful when you might have multiple forms on the page and you want to use ```form.errors```  or ```form.posted_successfully?```. Currently, you cannot identify which form actually has the errors.  If you try to use those properties, both forms display errors.  It would be awesome, if you could do something like:
```liquid

{% form 'customer', id: 'SomeOtherForm' %}
  {% if form.errors and form.id == 'SomeOtherForm' %}
  [...errors for SomeOtherForm...]
  {% endif %}
{% endform %}


[...more content on the same page...]

{% form 'customer', id: 'ContactFooter' %}
  {% if form.errors and form.id == 'ContactFooter' %}
  [...errors for ContactFooter...]
  {% endif %}
{% endform %}
```

# Discussion
Is having ```form.id``` always equal to ```nil``` expected behaviour?

**Comment by naxorn** [comment]
No.  Most of my hacks rely on the URL query parameters including 'unique' information.  Of course, it would be fine if at least the URL always had the form id, such as "[url]?customer_posted=true#FORM-ID".  Unfortunately, it is not always there such as when you encounter reCAPTCHA (#1511).

On my site, I limited myself to only a single customer subscribe form.  Plus, I had to add javascript on the subscribe form to add back the URL hash when I see that the URL showed a 'customer' type form was posted.  That way the page would jump back to the form on success.

Perhaps, you could have one 'form' and the rest of the subscribe parts as just links to the main form?

**Comment by eloyesp** [comment]
I'm having the same issue, it seems that it is not working for all the forms. I've just found that the product form use an id that is "product_form_{{ product.id }}". May be there is a similar trick for the rest of the forms.
